### PR TITLE
Fix prover specs for aptos_framework

### PR DESCRIPTION
**Pull request by macalinao**
### Description

Specs reference unresolved targets. Is there a way we can check this in CI?

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3460)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/genesis.move [hunk 1]
```diff
@@ -62,7 +62,19 @@ module aptos_framework::genesis {
...
+        let framework_reserved_addresses: vector<address> = vector::empty();
+        vector::push_back(&mut framework_reserved_addresses, @0x2);
+        vector::push_back(&mut framework_reserved_addresses, @0x3);
+        vector::push_back(&mut framework_reserved_addresses, @0x4);
+        vector::push_back(&mut framework_reserved_addresses, @0x5);
+        vector::push_back(&mut framework_reserved_addresses, @0x6);
+        vector::push_back(&mut framework_reserved_addresses, @0x7);
+        vector::push_back(&mut framework_reserved_addresses, @0x8);
+        vector::push_back(&mut framework_reserved_addresses, @0x9);
+        vector::push_back(&mut framework_reserved_addresses, @0x10);
```
Comment (billythedummy) [inline aptos-move/framework/aptos-framework/sources/genesis.move:76]: i just popped by randomly from twitter and i dont know jack about this codebase but i recently ran into a similar bug, so: is this supposed to be `@0xA` instead?

**Comment by wrwg** [comment]
Ian thanks for giving this a try and proposing a fix. My inclination is we would be to better fix the prover instead of making our code worse ;-) It is not too hard, here is the error in action (@junkil-park @meng-xu-cs) 

```
 aptos move prove
thread 'tokio-runtime-worker' panicked at 'not implemented: Not yet supported constant vector type: Vector(Primitive(Address))', /Users/wrwg/.cargo/git/checkouts/move-0639dd674f581c30/5ecf4df/language/move-prover/bytecode/src/stackless_bytecode_generator.rs:1274:30

```

File: aptos-move/framework/aptos-framework/sources/genesis.move (hunk 1 above)
Comment (wrwg) [reply aptos-move/framework/aptos-framework/sources/genesis.move:76]: Haha! I think you are totally right, this was intended to be `@0xa`

**Comment by meng-xu-cs** [comment]
Thanks Ian for the fix. Indeed we are aware of the problems. @junkil-park had two PRs on them:
https://github.com/aptos-labs/aptos-core/pull/3457
https://github.com/move-language/move/pull/411
And we will try to get them landed soon.

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/genesis.move [hunk 2]
```diff
@@ -62,7 +62,19 @@ module aptos_framework::genesis {
         aptos_governance::store_signer_cap(&aptos_framework_account, @aptos_framework, aptos_framework_signer_cap);
 
         // put reserved framework reserved accounts under aptos governance
-        let framework_reserved_addresses = vector<address>[@0x2, @0x3, @0x4, @0x5, @0x6, @0x7, @0x8, @0x9, @0x10];
+
+        // TODO: const vectors are not supported yet in the Move prover
+        let framework_reserved_addresses: vector<address> = vector::empty();
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/genesis.move:67]: This looks a bit ugly IMO but will leave it to @junkil-park and @meng-xu-cs regarding whether we can fix this on the prover side.

**Review by movekevin — CHANGES_REQUESTED** [review]

**Comment by macalinao** [comment]
> Thanks Ian for the fix. Indeed we are aware of the problems. @junkil-park had two PRs on them: #3457 [move-language/move#411](https://github.com/move-language/move/pull/411) And we will try to get them landed soon.

nice-- looking forward to seeing those get merged

File: aptos-move/framework/aptos-framework/sources/genesis.move (hunk 2 above)
Comment (wrwg) [reply aptos-move/framework/aptos-framework/sources/genesis.move:67]: @movekevin can you please look at the PR history before approving? A better fix for the core problem is on its way.

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/genesis.move:67]: Sorry I misread that comment as referring a different part of the change. But yes totally agree 👍

**Pull request closed without merging**